assets/src/edit-story/components/canvas/gridview/useGridViewKeys.js

```diff
@@ -0,0 +1,278 @@
+
+          if (canArrange) {
+            arrangePage({ pageId: focusedPageId, position: nextIndex });
+
+            // Focus on DOM element where this page is moving to
+            const page =
+              pageRefs.current && pageRefs.current[pageIds[nextIndex]];
+
+            focusOnPage(page);
+          }
+
+          break;
+        }
+        case 'ArrowUp':
+        case 'ArrowDown': {
+          const {
+            rows: numRows,
+            columns: numColumns,
+          } = getGridColumnAndRowCount(gridRef.current);
+          const currentIndex = pageIds.indexOf(focusedPageId);
+          const dir = key === 'ArrowDown' ? 1 : -1;
+
+          const currentRow = getRow(currentIndex, numColumns);
+          const currentColumn = getColumn(currentIndex, numColumns);
+
+          const nextIndex = getIndex(
+            currentRow + dir,
+            currentColumn,
+            numRows,
+            numColumns,
+            pageIds.length
+          );
+
+          if (nextIndex < 0) {
+            return;
+          }
+
+          const canArrange =
+            nextIndex !== currentIndex &&
+            nextIndex >= 0 &&
+            nextIndex <= pageIds.length - 1;
+
+          if (canArrange) {
+            arrangePage({ pageId: focusedPageId, position: nextIndex });
+
+            // Focus on DOM element where this page is moving to
+            const page =
+              pageRefs.current && pageRefs.current[pageIds[nextIndex]];
+
+            focusOnPage(page);
+          }
+
+          break;
+        }
+
+        default:
+          break;
+      }
+    },
+    [
+      currentPageId,
+      pageIds,
+      isRTL,
+      gridRef,
+      pageRefs,
+      focusedPageId,
+      arrangePage,
+    ]
+  );
+}
+
+// @todo: provide a cleaner `focusFirst()` API and takes into account
+// `tabIndex`, `disabled`, links, buttons, inputs, etc.
+function focusOnPage(page) {
+  const button = page?.querySelector('button');
+  if (button) {
+    button.focus();
+  }
+}
+
+function getArrowDir(key, pos, neg, isRTL) {
+  const rtlDir = isRTL ? -1 : 1;
+  if (key === pos) {
+    return rtlDir;
+  }
+  if (key === neg) {
+    return -1 * rtlDir;
+  }
+  return 0;
+}
+
+export function getGridColumnAndRowCount(gridElement) {
+  const { gridTemplateColumns, gridTemplateRows } = getComputedStyle(
+    gridElement
+  );
+  return {
+    rows: gridTemplateRows.includes('none')
+      ? 0
+      : gridTemplateRows.trim().split(' ').length,
+    columns: gridTemplateColumns.includes('none')
+      ? 0
+      : gridTemplateColumns.trim().split(' ').length,
+  };
+}
+
+// will return a 1 based index
+export function getRow(index, numColumns) {
+  return Math.ceil((index + 1) / numColumns);
+}
+
+// will return a 1 based index
+export function getColumn(index, numColumns) {
+  return (index % numColumns) + 1;
+}
+
+// will return a 0 based index
+export function getIndex(row, column, numRows, numColumns, numItems) {
```

Comment:
Since you have so many props you're passing here thoughts on making this an object so that the ordering isn't important? `getIndex({row, column, numRows, numColumns, numItems}) `